### ISSUE TITLE
508-defect-2 Fast Follow! [COGNITION]: Downloaded files SHOULD be consistent with what the user clicked to download

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#12889]
# [508-defect-2 Fast Follow!](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2) 

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

For the downloaded PDF upon a failed submission, we need to ensure that the downloaded file has a unique name that can be understood by the user. Currently, each download is named "92cbb5a2-db9c-4908-b3e0-dffef030d265.pdf" (or similar). Since the downloaded filename doesn't resemble the information the user clicked they may not be able to locate the file. Might we change the Content-Disposition HTTP header's filename parameter so that downloads will use the download attribute value, and the filename will align with what the user expects they are downloading?

For the stressful scenario of a person downloading failed application, it should be helpful if they can easily find the PDF. The name could be "10-10CG-SurnameOfApplicant.pdf".

@camerontesterman on the BAM2 team just worked on this for their Debt Letters work, and can likely guide the way. 

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a user, I want to know what document I am downloading is and recognize the file I downloaded.

## Environment

- Operating System: all
- Browser: all
- Server destination: staging

## Steps to Recreate

1. Complete the application
1. Reach the screen with the notice that something went wrong, with the prompt to download the application
1. Download the pdf, verify the name is a string of meaningless (to a user) characters

## Possible Fixes (optional)

1. Remove the download attribute
2. On the backend, set the file name to follow the pattern: @kevinmirc — I need to understand what meta data is available to create a meaningful file name.
3. Add type="application/pdf" to the link

## WCAG or Vendor Guidance (optional)

* [David Walsh - the download attribute](https://davidwalsh.name/download-attribute)

* [Chrome Dev Resources - Downloading resources in HTML5: a[download]](https://developers.google.com/web/updates/2011/08/Downloading-resources-in-HTML5-a-download)

* [MDN Content-Disposition](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition)

## Screenshots or Trace Logs

![Screen Shot 2020-08-26 at 12 58 38 PM](https://user-images.githubusercontent.com/57469/91333682-e2d86100-e79b-11ea-9a63-9830fe08f006.png)

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

**Comment by kevinmirc** [comment]
Hi @jenstrickland, Great catch! It would definitely be confusing to see that filename as a user.

On the backend, we currently return the header "Content-Disposition" with a suggested filename of "10-10CG_{veteran_first_name}_{veteran_last_name}.pdf" and a "Content-Type" of "application/pdf". So this may require updates on the front-end.

Here's an example:
```
"Content-Type": "application/pdf",
"Content-Disposition": "attachment; filename=\"10-10CG_Joan_Jerde.pdf\"; filename*=UTF-8''10-10CG_Joan_Jerde.pdf",
```

Another note about the filename, this is the same format the attachment stored in CARMA and that CSCs expect. But if there is a way to make it more user-friendly please let me know.

@jenstrickland Will this filename format suffice? 
@LinnJS Can we use the suggested filename from the API response header?

**Comment by kevinmirc** [comment]
Hi @jenstrickland,

Wanted to followup on this issue. It should be resolved with this PR: https://github.com/department-of-veterans-affairs/vets-website/pull/14029

**Comment by ala-yna** [comment]
Hi @jenstrickland ! Any chance you could help us validate and close?

**Comment by jenstrickland** [comment]
🎉  Yes! `10-10CG_a_b.pdf` was the downloaded file name for Veteran A B.
